example/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"context"
+
+	"github.com/carousell/go-logging"
+	"github.com/carousell/go-logging/gokit"
+)
+
+func main() {
+
+	ctx := context.Background()
+
+	logging.RegisterLogger(gokit.NewLogger())
+
+	ctx = logging.WithLogger(ctx,
+		logging.WithAttr("main1", "main1"),
+		logging.WithAttr("main2", 123),
+	)
+
+	logging.InfoV2(ctx, "before calling foo()")
+	//{"@timestamp":"2023-04-27T00:39:45.573743+08:00","caller":"go-logging/new.go:33","level":"info","main1":"main1","main2":123,"msg":"before calling foo()"}
+
+	foo(ctx)
+	logging.InfoV2(ctx, "after called foo()")
+	//{"@timestamp":"2023-04-27T00:39:45.574079+08:00","caller":"go-logging/new.go:33","level":"info","main1":"main1","main2":123,"msg":"after called foo()"}
```

Comment:
- attrs from current scope: "main1", "main2"

NOTE that this log appears after calling `foo(ctx)`, but the attrs within `foo(ctx)` does not leak out